src/libYARP_pcl/src/CMakeLists.txt

```diff
@@ -20,9 +20,11 @@ target_compile_features(YARP_pcl INTERFACE cxx_std_14)
 target_link_libraries(YARP_pcl INTERFACE YARP::YARP_sig)
 list(APPEND YARP_pcl_PUBLIC_DEPS YARP_sig)
 
-target_include_directories(YARP_pcl SYSTEM INTERFACE ${PCL_INCLUDE_DIR})
-target_link_libraries(YARP_pcl INTERFACE ${PCL_LIBRARIES})
-list(APPEND YARP_pcl_PUBLIC_DEPS PCL)
+target_include_directories(YARP_pcl SYSTEM INTERFACE ${PCL_COMMON_INCLUDE_DIR}
+                                                     ${PCL_IO_INCLUDE_DIR})
```

Comment:
On PCL 1.10 (using CMake targets), `INTERFACE_INCLUDE_DIRECTORIES` is defined as `"${PCL_${COMPONENT}_INCLUDE_DIRS};${PCL_CONF_INCLUDE_DIR}"`. Not only we are missing the `PCL_CONF_INCLUDE_DIR` part (I don't think this is critical here, anyway), but also note the trailing `S` in the component's include dir variable:
* `${PCL_XXX_INCLUDE_DIR}`: just this component's include header dir(s)
* `${PCL_XXX_INCLUDE_DIRS}`: same as above **plus** dependencies

On PCL 1.8 (Ubuntu bionic), my sample application (see linked issue) throws a compile error when using `PCL_XXX_INCLUDE_DIR` due to an unresolved Eigen dependency.

I think this is better written simply as `${PCL_INCLUDE_DIRS}`, thus accounting for all internal and external dependencies (pcl_io also dependes on pcl_octree).

Reply:
reverted, see b0443ce491b6793746eb8fcb0a586b6e2d2f4b1f